Add hint to add EntityCustomFieldsTrait (#795)
shopware/docs: guides/plugins/plugins/framework/custom-field/add-custom-field.md

This short section will cover how to add a custom field support for your custom entity. As previously mentioned, the example from our [Add custom complex data](../data-handling/add-custom-complex-data.md) guide is used and extended here.

> In order to support custom fields with your custom entity, there are three steps necessary:
> 
> * Add `EntityCustomFieldsTrait` trait to your `Entity`
> * Add a `CustomFields` field to your `EntityDefinition`
> * Add a column `custom_fields` to your entities' database table via migration

Reviewer: ```suggestion
In order to support custom fields with your custom entity, there are three necessary steps :

* Add `EntityCustomFieldsTrait` trait to your `Entity`.
* Add a `CustomFields` field to your `EntityDefinition`.
* Add a column `custom_fields` to your entities' database table via migration.
```

Also, you may want to add translatable custom fields, which is also covered in very short here.

### Add custom field to entity

{% hint style="info" %}
  Available starting with Shopware 6.4.1.0.
{% endhint %}

> Let's assume you already got a working and running entity definition.
> If you want to support custom fields with your custom entity,
> you may add the `EntityCustomFieldsTrait` to your entity class, so the methods
> `getCustomFields()` and `setCustomFields()` can be used.

Reviewer: ```suggestion
Let's assume you already got a working and running entity definition. If you want to support custom fields with your custom entity, you may add the `EntityCustomFieldsTrait` to your entity class, so the methods `getCustomFields()` and `setCustomFields()` can be used.
```

{% code title="<plugin root>/src/Core/Content/Example/ExampleEntity.php" %}

```php
use Shopware\Core\Framework\DataAbstractionLayer\Entity;
use Shopware\Core\Framework\DataAbstractionLayer\EntityCustomFieldsTrait;
use Shopware\Core\Framework\DataAbstractionLayer\EntityIdTrait;

[...]
class ExampleEntity extends Entity
{
    use EntityIdTrait;
    use EntityCustomFieldsTrait;

    [...]

}
```

{% endcode %}

### Add custom field to entity definition

> Now follows the important parts. For this to work at all, you have to add the
> Data Abstraction Layer \(DAL\) field `CustomFields` to your entity definition.

Reviewer: ```suggestion
Now follows the important part. For this to work, you have to add the Data Abstraction Layer \(DAL\) field `CustomFields` to your entity definition.
```
